libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -367,6 +367,7 @@ bool HTTPClient::begin(String host, uint16_t port, String uri, const uint8_t htt
  */
 void HTTPClient::end(void)
 {
+    _canReuse = false;
```

Comment:
Even if the user calls `::end`, _client is never cleared in this code so the object will always print the debug message nastygram when you destroy the object.  Suggest the simplest thing here is to set _client to NULL at this point in the code.

I would also suggest setting _tcpDeprecated to NULL here as well, since that will take care of deletion when the emulated object is no longer needed.

Reply:
::end() calls ::disconnect(). If not connected both pointers are already NULL, otherwise _client will be set to nullptr and _tcpDeprecated will be reset to nullptr. Agree?

Reply:
Ah, got it.  Didn't catch that since disconnect() isn't part of the diff so it's not shown, but checking the full code it looks OK.